src/main/java/org/folio/circulation/support/Interval.java

```diff
@@ -0,0 +1,204 @@
+package org.folio.circulation.support;
+
+import static java.time.Duration.ofMillis;
+import static java.time.Instant.ofEpochMilli;
+import static java.time.ZoneOffset.UTC;
+import static org.folio.circulation.support.utils.ClockUtil.getInstant;
+import static org.folio.circulation.support.utils.ClockUtil.getLocalDateTime;
+import static org.folio.circulation.support.utils.ClockUtil.getZoneId;
+import static org.folio.circulation.support.utils.DateTimeUtil.isAfterMillis;
+import static org.folio.circulation.support.utils.DateTimeUtil.isBeforeMillis;
+import static org.folio.circulation.support.utils.DateTimeUtil.isSameMillis;
+import static org.folio.circulation.support.utils.DateTimeUtil.millisBetween;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+import org.folio.circulation.support.utils.DateTimeUtil;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Value;
+import lombok.With;
+
+/**
+ * A simple time interval using an inclusive begin and a duration.
+ * <p>
+ * The begin represents a zoned date and time in which the interval
+ * begins, inclusively.
+ * <p>
+ * The duration represents how long from the begin date and time.
+ * <p>
+ * This is immutable.
+ */
+@AllArgsConstructor
+@Builder
+@Value
+@With
+public class Interval {
```

Comment:
What is this needed for?

Reply:
To jog you memory, several months ago [I presented this problem in the CIRC-966 Jira ticket](https://issues.folio.org/browse/CIRC-966?focusedCommentId=101022&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-101022).

Reply:
Is there a plan to remove it later?

Reply:
> JavaTime got rid of `Interval` that JodaTime had.
> This is a recreation to avoid rewriting all of the code that relied on the `Interval` class.
> 
> I would love to not need this, but getting rid of this would require even more changes than already exist.

Ok. Changing the code to need this is out of the scope of this work and so this is required.

Have I understood this correctly?

Reply:
```Have I understood this correctly?```
Yes.

I think I looked into this several months ago and backed away once I realized what would be involved.